core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -56,7 +60,9 @@ class LogManager(val logDirs: Array[File],
                  scheduler: Scheduler,
                  val brokerState: BrokerState,
                  brokerTopicStats: BrokerTopicStats,
-                 time: Time) extends Logging {
+                 val zkUtils: ZkUtils,
```

Comment:
@junrao BTW, here is the current workflow if `LogManger.handleLogDirFailure()` is called by `LogCleaner`.

- `LogManager.handleLogDirFailure()` will remove this log dir and partitions and inform controller via zookeeper.
- Controller sends LeaderAndIsrRequest for all partitions on this broker
- `Partition.getOrCreateReplica()` for partitions on the offline log directory will throw KafkaStorageException, which in turn triggers `ReplicaManager.handleLogDirFailure()` so that those partitions can be removed from `ReplicaManager` and the `ReplicaFetcherManager`.